search/engine.go

```diff
@@ -406,10 +483,30 @@ func joinResponses(r1 *elastic.SearchResult, r2 *elastic.SearchResult, sortBy st
 func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, from int, size int, preference string) (interface{}, error) {
 	log.Infof("Query: %+v sort by: %s, from: %d, size: %d", query, sortBy, from, size)
 	multiSearchService := e.esc.MultiSearch()
-	// Content Units
-	AddContentUnitsSearchRequests(multiSearchService, query, sortBy, 0, from+size, preference)
-	// Collections
-	AddCollectionsSearchRequests(multiSearchService, query, sortBy, 0, from+size, preference)
+
+	var indexesCount int
+	if len(query.Filters) == 0 {
+		log.Info("No section selected. Searching in all indexes.")
+		indexesCount = 3
+	} else if _, ok := query.Filters[consts.FILTERS[consts.FILTER_SECTION_SOURCES]]; ok {
+		log.Info("Sources section selected. Searching only in sources.")
+		indexesCount = 1
+	} else {
+		log.Info("Non sources section selected. Searching only in content units and collections.")
+		indexesCount = 2
+	}
+
+	if indexesCount >= 2 {
```

Comment:
This is not the right way to handle this we have 3 states:

NO_FILTER
ONLY_SOURCES
WITHOUT_SOURCES

You need an enum with those values. Also I have noticed that in the client we should consider disabling non-valid filtering...

Reply:
```
requests = make([]...)
if NO_FILTER || WITHOUT_SOURCES
  requests = append(requests,
      content units,
      collections)  // I think append is variadic

if NO_FILTER || ONLY_SOURCES
  requests = append(requests, sources)
```

Reply:
Issue created AS-36